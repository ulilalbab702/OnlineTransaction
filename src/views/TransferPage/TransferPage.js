import React, { Component } from 'react'
import {
  Row,
  Col,
  Button,
} from 'reactstrap';
import {
  DropIcon
} from "assets/icons";
import "./style/index.scss";
import { IconTimer } from 'assets/icons';
import Breadcrumb from '../../components/Breadcrumb/Breadcrumb';
import { formatCurrency, formatVaNumber } from "utils/format.helper";
import moment from "moment";
import firebase from "../../firebase/firebase";
import { MENU } from 'constants/menu';


class TransferPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      changeStatus: 3,
      dataPayment: this.props?.history?.location?.state ? this.props?.history?.location?.state : null,
      isCopied: false,
    }
  };

  setAnalyticClevertap = (action, event, screen, product) => {
    if (this.props.user) {
      const { userName } = this.props.user;
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_LogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_LogedIn"}`, this.props.user);
      }
      firebase.analytics().setUserId(userName);
      firebase.analytics().setUserProperties(userName, "username:" + userName + `||View: ${screen}`);
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_LogedIn"}`);
      firebase.analytics().setUserProperties("username", userName);
      firebase.analytics().setUserProperties(action, `${event + "_LogedIn"}`);
    }
  };

  async componentDidMount() {
    this.setAnalyticClevertap("view", "View_PaymentDetail_Screen", "View_PaymentDetail_Screen", null);
    window.scrollTo(0, 0);
  };

  handleCopyVaNumber = () => {
    navigator.clipboard.writeText(this.state.dataPayment?.paymentNumber);
    this.setState({ isCopied: true });
    this.setAnalyticClevertap("click", "Click_CopyVaNumber", "View_PaymentDetail_Screen", null);
  };

  _renderBreadcrumb = () => {
    const breadcrums = [
      {
        'url': MENU.HOME,
        'name': 'Part Online Transaction'
      }
    ];
    return (
      <Breadcrumb
        linkBreadcrumb={breadcrums}
        typography={"Detail Pembayaran"}
      />
    )
  };

  render() {
    return (
      <Col className='paddingTP'>
        <Col className='breadCrumbTP'>
          {this._renderBreadcrumb()}
        </Col>
        <h3 className="styleTPBold">Pembayaran</h3>
        <Row className="rowTotalPayment">
          <p className="styleTPregular">Total Pembayaran</p>
          <p className="fontPriceTP">RP {formatCurrency(this.state.dataPayment?.grandTotal)}</p>
        </Row>
        <hr className="mt-0"></hr>
        <Col lg="12" md="12" style={{ padding: '0' }}>
          <p className="styleTPregular">{this.state.dataPayment?.paymentMethod}</p>
          <p className="styleTPregular1">Virtual Account Number</p>
          <div style={{ display: 'flex', flexDirection: 'row' }}>
            <p className="fontRekTP">{formatVaNumber(this.state.dataPayment?.paymentNumber)}</p>
            <p
              className="copyTeks"
              onClick={() => this.handleCopyVaNumber()}
            >{this.state.isCopied ? "Berhasil disalin." : "Salin"}</p>
          </div>
          <p className="styleTPCek">Akan dicek dalam 30 menit setelah pembayaran berhasil</p>
          <p className='styleTPKet'>
            Periksa informasi yang tertera di layar. Pastikan total tagihan dan username sudah benar.
          </p>
          <Row className="rowTimer">
            <img className='iconTimer' src={IconTimer}></img>
            <p className='styleTimer'>Bayar Sebelum {moment(this.state.dataPayment?.paymentExpired + "Z").format("DD MMMM YYYY, HH:mm")} WIB</p>
          </Row>
          <Row className="rowBtnTP">
            <Button style={{ backgroundColor: "#FFD500", border: "none", color: '#000000', float: 'right' }} className="btnOkTP"
              onClick={() => {
                this.setAnalyticClevertap("click", "Click_OK", "View_PaymentDetail_Screen", null);
                this.props.push(MENU.LIST_ORDER, this.state.changeStatus)
              }}
            >OK</Button>
          </Row>
        </Col>
      </Col>
    )
  }
}

export default TransferPage;