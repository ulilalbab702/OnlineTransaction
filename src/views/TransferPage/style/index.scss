@import url("./desktop/TransferPageStyle.scss");
@import url("./tablet/TransferPageStyle.scss");
@import url("./mobile/TransferPageStyle.scss");
