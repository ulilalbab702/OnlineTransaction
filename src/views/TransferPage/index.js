import TransferPage from './TransferPage';
import { connect } from 'react-redux';
import { LOGIN_TYPE } from 'actions/actionTypes';
import { login } from 'actions/user';
import { push } from 'connected-react-router';
import {
    createLoadingSelector,
} from 'utils/selector.helper'

//Set selector loading 
const loadingSelector = createLoadingSelector([LOGIN_TYPE]);
const mapStateToProps = state => ({
    ...state.user,
    isLoading: loadingSelector(state),
});

const mapDispatchToProps = dispatch => ({
    login: (data) => dispatch(login(data)),
    push: (url) => dispatch(push(url)),
});

export default connect(mapStateToProps, mapDispatchToProps)(TransferPage);