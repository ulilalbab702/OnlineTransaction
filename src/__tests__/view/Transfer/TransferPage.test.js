import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TransferPage from "views/TransferPage/TransferPage";

Enzyme.configure({ adapter: new Adapter() });
describe("Transfer Page Component", () => {
    const wrapper = shallow(
        <TransferPage />
    );
    describe("Transfer Page Component", () => {
        it("Should match to snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});