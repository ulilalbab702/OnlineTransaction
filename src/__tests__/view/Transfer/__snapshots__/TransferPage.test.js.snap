// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transfer Page Component Transfer Page Component Should match to snapshot 1`] = `
<div
  style={
    Object {
      "marginBottom": "4rem",
    }
  }
>
  <Row
    style={
      Object {
        "marginRight": "0",
      }
    }
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="bgTP"
      lg="8"
      md="10"
      sm="10"
      style={
        Object {
          "paddingLeft": "0",
        }
      }
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <h3
        className="styleTPBold"
        style={
          Object {
            "marginTop": "2rem",
          }
        }
      >
        Pembayaran
      </h3>
      <Row
        style={
          Object {
            "marginTop": "2.5rem",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Col
          lg="6"
          md="6"
          sm="6"
          style={
            Object {
              "paddingRight": "0",
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <h4
            className="styleTPregular"
          >
            Total Pembayaran
          </h4>
        </Col>
        <Col
          lg="6"
          md="6"
          sm="6"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="6"
        >
          <h3
            className="fontPriceTP"
          >
            RP 10.500.000
          </h3>
        </Col>
      </Row>
      <hr
        className="mt-0"
      />
      <Col
        lg="12"
        md="12"
        style={
          Object {
            "padding": "0",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="marginTP1"
        >
          <h3
            className="styleTPregular"
            style={
              Object {
                "fontWeight": "600",
              }
            }
          >
            Bank Mandiri (Dicek Otomatis)
          </h3>
        </div>
        <div
          className="marginTP1"
          style={
            Object {
              "marginBottom": "0",
            }
          }
        >
          <h4
            className="styleTPregular"
            style={
              Object {
                "fontSize": "16px",
              }
            }
          >
            No. Rekening:
          </h4>
        </div>
        <div
          className="marginTP1"
          style={
            Object {
              "marginBottom": "0.2rem",
              "marginTop": "0",
            }
          }
        >
          <h3
            className="fontRekTP"
          >
            600 0812 1831 0120
          </h3>
        </div>
        <div
          className="styleTPCek"
        >
          Dicek dalam 10 menit setelah pembayaran berhasil
        </div>
        <Col
          className="styleTPKet"
          style={
            Object {
              "padding": "0",
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Bayar pesanan ke Virtual Account di atas sebelum membuat pesanan kembali dengan Virtual Account agar nomor tetap sama. Hanya menerima dari Bank Mandiri
        </Col>
        <hr />
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            lg="12"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="containerTP"
              onClick={[Function]}
            >
              <div
                className=" TPord flex-containerTP"
              >
                <h3
                  className="styleTP"
                >
                  Petunjuk Transfer mBanking
                </h3>
              </div>
              <div
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <img
                  className="dropIcon"
                  src="dropIcon.png"
                />
              </div>
            </div>
          </Col>
        </Row>
        <hr
          style={
            Object {
              "marginTop": "1.5rem",
            }
          }
        />
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            lg="12"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="containerTP"
              onClick={[Function]}
            >
              <div
                className=" TPord flex-containerTP"
              >
                <h3
                  className="styleTP"
                >
                  Petunjuk Transfer iBanking
                </h3>
              </div>
              <div
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <img
                  className="dropIcon"
                  src="dropIcon.png"
                />
              </div>
            </div>
          </Col>
        </Row>
        <hr
          style={
            Object {
              "marginTop": "1.5rem",
            }
          }
        />
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            lg="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              className="btnOkTP"
              color="secondary"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#FFD500",
                  "border": "none",
                  "color": "#000000",
                  "float": "right",
                }
              }
              tag="button"
            >
              OK
            </Button>
          </Col>
        </Row>
      </Col>
    </Col>
  </Row>
</div>
`;
